bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -749,13 +770,27 @@ public synchronized ModuleHandlerFactory addingService(ServiceReference/* <Modul
             }
         }
         if (notInitailizedRules != null) {
-            for (String rUID : notInitailizedRules) {
-                setRule(rUID);
+            for (final String rUID : notInitailizedRules) {
+                scheduleRuleInitialization(rUID);
             }
         }
         return mhf;
     }
 
+    private synchronized void scheduleRuleInitialization(final String rUID) {
+        Future f = scheduleTasks.get(rUID);
+        if (f == null) {
+            ScheduledExecutorService ex = getScheduledExecutor();
+            f = ex.schedule(new Runnable() {
+                @Override
+                public void run() {
+                    setRule(rUID);
+                }
+            }, 500, TimeUnit.MILLISECONDS);
+            scheduleTasks.put(rUID, f);
+        }
```

Comment:
Yes you are right, I'm not sure that it is the last try, but if in the else statement the timer is restarted it will not change anything. In your use case with 30 sec start we  will have 60 tries again. Even more the rule will always wait 500 ms after last try in which it is initialized.